File: test-suite/src/test/java/io/micronaut/kafka/docs/admin/MyTopicFactoryTest.java
package io.micronaut.kafka.docs.admin;

import io.micronaut.configuration.kafka.admin.KafkaNewTopics;
import io.micronaut.context.ApplicationContext;
import org.junit.jupiter.api.Test;

import java.util.Map;
import java.util.concurrent.ExecutionException;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;
import static org.junit.jupiter.api.Assertions.assertEquals;

class MyTopicFactoryTest {

    @Test
    void testNewTopics() throws ExecutionException, InterruptedException {
        try (ApplicationContext ctx = ApplicationContext.run(Map.of(
            "kafka.enabled", "true",
            "spec.name", "MyTopicFactoryTest"
        ))) {
            final KafkaNewTopics newTopics = ctx.getBean(KafkaNewTopics.class);
            await().atMost(5, SECONDS).until(() -> areNewTopicsDone(newTopics));
            assertEquals(1, newTopics.getResult().numPartitions("my-new-topic-1").get());
            assertEquals(2, newTopics.getResult().numPartitions("my-new-topic-2").get());
        }
    }

    // tag::result[]
    boolean areNewTopicsDone(KafkaNewTopics newTopics) {
        return newTopics.getResult().all().isDone();
    }
    // end::result[]
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/admin/MyTopicFactoryTest.kt
package io.micronaut.kafka.docs.admin

import io.micronaut.configuration.kafka.admin.KafkaNewTopics
import io.micronaut.context.ApplicationContext
import org.awaitility.Awaitility
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.util.concurrent.ExecutionException
import java.util.concurrent.TimeUnit

internal class MyTopicFactoryTest {

    @Test
    @Throws(ExecutionException::class, InterruptedException::class)
    fun testNewTopics() {
        ApplicationContext.run(
            mapOf(
                "kafka.enabled" to "true",
                "spec.name" to "MyTopicFactoryTest"
            )
        ).use { ctx ->
            val newTopics = ctx.getBean(KafkaNewTopics::class.java)
            Awaitility.await().atMost(5, TimeUnit.SECONDS).until { areNewTopicsDone(newTopics) }
            val result = newTopics.result.orElseThrow()
            assertEquals(1, result.numPartitions("my-new-topic-1").get())
            assertEquals(2, result.numPartitions("my-new-topic-2").get())
        }
    }

    // tag::result[]
    fun areNewTopicsDone(newTopics: KafkaNewTopics): Boolean {
        return newTopics.result.map { it.all().isDone }.orElse(false)
    }
    // end::result[]
}
